src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratorIdentity.cs

```diff
@@ -3,41 +3,50 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
 using Microsoft.CodeAnalysis.Diagnostics;
 
-namespace Microsoft.CodeAnalysis
+namespace Microsoft.CodeAnalysis;
+
+/// <remarks>
+/// Assembly path is used as a part of a generator identity to deal with the case that the user accidentally installed the same
+/// generator twice from two different paths, or actually has two different generators that just happened to use the same name.
+/// In the wild we've seen cases where a user has a broken project or build that results in the same generator being added twice;
+/// we aren't going to try to deduplicate those anywhere since currently the compiler does't do any deduplication either:
+/// you'll simply get duplicate outputs which might collide and cause compile errors. If https://github.com/dotnet/roslyn/issues/56619
+/// is addressed, we can potentially match the compiler behavior by taking a different approach here.
+/// </remarks>
+[DataContract]
+internal readonly record struct SourceGeneratorIdentity
 {
-    /// <remarks>
-    /// Assembly path is used as a part of a generator identity to deal with the case that the user accidentally installed the same
-    /// generator twice from two different paths, or actually has two different generators that just happened to use the same name.
-    /// In the wild we've seen cases where a user has a broken project or build that results in the same generator being added twice;
-    /// we aren't going to try to deduplicate those anywhere since currently the compiler does't do any deduplication either:
-    /// you'll simply get duplicate outputs which might collide and cause compile errors. If https://github.com/dotnet/roslyn/issues/56619
-    /// is addressed, we can potentially match the compiler behavior by taking a different approach here.
-    /// </remarks>
-    internal record struct SourceGeneratorIdentity
+    [DataMember(Order = 0)]
```

Comment:
Out of curiosity, why not the primary constructor style here too?

Reply:
i'll move to that.